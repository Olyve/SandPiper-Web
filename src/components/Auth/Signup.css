.signup {
  margin: 10px;
}

.submit {
  display: inline-block;
}

form {
  vertical-align: center;
}

label input {
  margin-bottom: 10px;
  margin-left: 10px;
}