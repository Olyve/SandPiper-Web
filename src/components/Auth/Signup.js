import React, { Component } from 'react';
import './Signup.css';

class Signup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(event) {
    let target = event.target;
    let value = target.value;
    let name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    this.props.handleSignup({ ...this.state });

    event.preventDefault();
  }

  render() {
    return (
      <div className='signup'>
        <form>
          <label>Email: 
          <input type='text' name='email' onChange={this.onChange}/>
          </label>
          <br />
          <label>Password: 
          <input type='password' name='password' onChange={this.onChange}/>
          </label>
          <br />
          <button className='submit' onClick={this.handleSubmit}>Submit</button>
        </form>
      </div>
    );
  }
}

export default Signup;