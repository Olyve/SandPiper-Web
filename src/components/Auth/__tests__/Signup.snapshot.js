import React from 'react';
import renderer from 'react-test-renderer';
import Signup from '../Signup';

describe('Signup', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(
      <Signup />
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});