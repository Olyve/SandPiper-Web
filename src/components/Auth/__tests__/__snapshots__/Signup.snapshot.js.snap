// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should render without crashing 1`] = `
<div
  className="signup"
>
  <h2>
    Register New User
  </h2>
  <form>
    <label>
      Email:
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Password:
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <br />
    <button
      className="submit"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
